### PR TITLE
Merge in Yost line

### PR DESCRIPTION
**Pull request by danrue**
Merging in a long lived branch from Aug 2015.

Changes should be backward compatible, but the minor version will be bumped since they are considerable. Changes include:
- Remove cruft (.kitchen.yml, metadata.yml, Dockerfile.j2, etc)
- Improve handlers to make supervisor restarts actually work (!)
- Simplify testing interface, and expand the set of tests running
- Support multiple programs in a backward compatible way, by adding the supervisor_programs list.

Supercedes #6 #4

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,7 +1,3 @@
 ---
-
-supervisor_version: 3.0b2-1
```
Comment (danrue) [inline defaults/main.yml]: version no longer supported. just get whatever is default via apt. (that's probably safer i reckon)

File: handlers/main.yml [hunk 2]
```diff
@@ -1,10 +1,25 @@
 ---
+# handlers file for example
```
Comment (eggilbert) [inline handlers/main.yml]: Can we nuke this? It's not really an example since you can't override handlers.

File: handlers/main.yml [hunk 3]
```diff
@@ -1,10 +1,25 @@
...
-    state: restarted
-    sleep: "{{ supervisor_restart_sleep | default(omit) }}"
-  register: supervisor_restarted
-  until: supervisor_restarted|success
-  retries: 1
-  delay: 5
+  command: /bin/true
+  notify:
+    - stop supervisor
+    - wait stop supervisor
```
Comment (eggilbert) [inline handlers/main.yml]: This seems like it might actually be reliable now!

File: templates/supervisor.conf.j2 [hunk 4]
```diff
@@ -1,31 +1,33 @@
...
+redirect_stderr={{item.supervisor_redirect_stderr|default('false')}}
+stdout_logfile={{item.supervisor_stdout_logfile |default('/var/log/supervisor/stdout.log') }}
+stdout_logfile_maxbytes={{item.supervisor_stdout_logfile_maxbytes|default('25MB')}}
+stdout_logfile_backups={{item.supervisor_stdout_logfile_backups|default('10')}}
+stdout_capture_maxbytes={{item.supervisor_stdout_capture_maxbytes|default('5MB')}}
+stderr_logfile={{item.supervisor_stderr_logfile|default('/var/log/supervisor/stderr.log')}}
+stderr_logfile_maxbytes={{item.supervisor_stderr_logfile_maxbytes|default('25MB')}}
+stderr_logfile_backups={{item.supervisor_stderr_logfile_backups|default('10')}}
+stderr_capture_maxbytes={{item.supervisor_stderr_capture_maxbytes|default('5MB')}}
 serverurl=AUTO
```
Comment (eggilbert) [inline templates/supervisor.conf.j2]: Just want to make sure this 1 line that didn't change in here is actually not supposed to change/move. Seems suspicious.

Comment (danrue) [inline templates/supervisor.conf.j2]: the old template changed to the .single.j2 template. The only line that changed between the old supervisor.conf.j2 and the new supervisor.conf.single.j2 is:

```
3c3
< process_name=%(program_name)s
---
> process_name={{supervisor_process_name}}
```

File: test.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
+---
+- hosts: test
+  sudo: true
+  vars:
+    ##we will use these in our tests otherwise could scope to role
```
Comment (eggilbert) [inline test.yml]: This comment is kinda confusing. It seems like right now we have both externally defined programs, as well as one slapped on the role invocation. Is that intentional to have both in the tests?

File: vars/main.yml [hunk 6]
```diff
@@ -1 +1,2 @@
 ---
+# vars file for example
```
Comment (eggilbert) [inline vars/main.yml]: Is this vars file even necessary?

File: test.yml (hunk 5 above)
Comment (danrue) [inline test.yml]: I didn't really look at the 6 test cases in tests/, i just made sure they ran and passed. It's calling the role once, and then exercising various handler conditions in the tests. I guess i don't get the comment either, but it seemed like some thought was put into these tests and the previous tests were naive, so i just went with it.

File: vars/main.yml (hunk 6 above)
Comment (danrue) [inline vars/main.yml]: I'll remove the directory and rerun tests.

**Comment by eggilbert** [comment]
Overall this seems like this should be more robust

File: handlers/main.yml (hunk 3 above)
Comment (eggilbert) [inline handlers/main.yml]: Re-read my submission from https://github.com/SPSCommerce/ansible-role-supervisor/pull/8 and it seems like the handlers here are accomplishing the same thing, but better.

File: test.yml (hunk 5 above)
Comment (eggilbert) [inline test.yml]: `¯\_(ツ)_/¯`

**Pull request merged**